cubeviz/layout.py

```diff
@@ -225,12 +227,27 @@ def _update_slice(self, index):
             image._widget.state.slices = (index, y, x)
 
         self.ui.text_slice.setText(str(index))
-        self.ui.text_wavelength.setText(str(self._wavelengths[index]))
+
+        # Get the wavelength units in order to set the wavelength value's number format
+        self.ui.text_wavelength.setText(self._wavelength_format.format(self._wavelengths[index]))
 
     def _enable_slider(self):
         self.ui.value_slice.setEnabled(True)
         self.ui.value_slice.setMinimum(0)
 
+        # Store the wavelength units and format
+        if self._wavelength_units == None:
```

Comment:
This check might be redundant since I'm not sure whether `self._wavelength_units` will ever be anything other than `None` when `_enable_slider` is called.

Reply:
good point. I removed it.